geomesa-index-api/src/test/scala/org/locationtech/geomesa/index/geotools/GeoMesaDataStoreTest.scala

```diff
@@ -68,5 +71,32 @@ class GeoMesaDataStoreTest extends Specification {
       val results = SelfClosingIterator(ds.getFeatureReader(query, Transaction.AUTO_COMMIT)).toSeq
       results must beEmpty
     }
+    "intercept and rewrite queries" in {
+      val sft = SimpleFeatureTypes.createType("rewrite", "name:String,age:Int,dtg:Date,*geom:Point:srid=4326")
+      sft.getUserData.put(SimpleFeatureTypes.Configs.QUERY_INTERCEPTORS, classOf[TestQueryInterceptor].getName)
+
+      val ds = new TestGeoMesaDataStore(true)
+      ds.createSchema(sft)
+
+      ds.getFeatureSource(sft.getTypeName).addFeatures(new ListFeatureCollection(sft, features.toArray[SimpleFeature]))
+
+      // INCLUDE should be re-written to EXCLUDE
+      ds.getQueryPlan(new Query(sft.getTypeName)) must beEmpty
+      var results = SelfClosingIterator(ds.getFeatureReader(new Query(sft.getTypeName), Transaction.AUTO_COMMIT)).toSeq
+      results must beEmpty
+
+      // other queries should go through as normal
+      results = SelfClosingIterator(ds.getFeatureReader(new Query(sft.getTypeName, ECQL.toFilter("bbox(geom,39,54,51,56)")), Transaction.AUTO_COMMIT)).toSeq
+      results must haveLength(10)
+    }
+  }
+}
+
+object GeoMesaDataStoreTest {
```

Comment:
How hard would it be to add a unit test demonstrating multiple query interceptors which clearly shows the order of operations?

Reply:
probably not too hard, but it's a `.foreach` so dunno if we need to check it so rigorously :)

Reply:
https://github.com/locationtech/geomesa/pull/2155/files#diff-70d7b065fbfde37dcee7f8ab3966d3cdR102